### PR TITLE
Added .editorconfig file

### PR DESCRIPTION
**Pull request by AurelioDeRosa**
`.editorconfig` file taken directly from jQuery's repository.

### CODE REVIEW

**Comment by scottgonzalez** [comment]
It's kinda silly to do a direct copy from a different project with different types of files and a different directory structure. Half of these rules are for files that don't even exist, and the majority of the files in this repo don't even get indentation rules applied to them.

**Comment by AurelioDeRosa** [comment]
For the benefit of solving this issue, it'd be great if you could point on specific lines that you'd like to change/add/remove. By doing so, we can have a change that works for everyone and  improve the _status quo_. In its current state your comment doesn't really help.

**Comment by kswedberg** [comment]
@AurelioDeRosa: why bother specifying `[**/.js]`, `[**.xml]`, etc.? Is there a type of file in which we don't use `indent_style = tab`? I think having one `indent_style = tab` along with the other rules should be sufficient. Or am I missing something?

**Comment by scottgonzalez** [comment]
> I think having one indent_style = tab along with the other rules should be sufficient. Or am I missing something?

Right, almost everything is tab indented, so having that be the default would make sense. `package.json` and `*.yml` should be set to two spaces though.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+root = true
+
+
+[*]
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+# Tabs in JS unless otherwise specified
```
Comment (scottgonzalez) [inline .editorconfig]: There is nothing else specified.

**Comment by mgol** [comment]
ping @AurelioDeRosa

**Comment by mgol** [comment]
@AurelioDeRosa It seems to me that [current version of Core's `.editorconfig`](https://github.com/jquery/jquery/blob/43a0b5e/.editorconfig) would be a good fit, the only modification needed is to apply 2 spaces to `*.yml` files as well as @scottgonzalez indicated.

Would you like to do it? If not, I'd close this PR and I can do it separately.

**Pull request closed without merging**